### PR TITLE
Issue #14137: Enable `MockitoMockClassReference` check

### PR DESCRIPTION
**Pull request by rickie**
Issue #14137.

This PR enables the https://error-prone.picnic.tech/bugpatterns/MockitoMockClassReference/ check.

### CODE REVIEW

**Review by romani — APPROVED** [review]
Ok to merge

Needs more reviews as there is code changes

File: pom.xml [hunk 1]
```diff
@@ -254,6 +254,7 @@
       -Xep:JUnitValueSource:OFF
       <!-- Until https://github.com/checkstyle/checkstyle/issues/14194. -->
       -Xep:LexicographicalAnnotationListing:OFF
+      -Xep:MockitoMockClassReference:ERROR
```
Comment (rnveach) [inline pom.xml:257]: I think there should be a clear separation between OFFs and ERRORs. You added this one right after an OFF with a reason for being OFF. At just a glance, it looks like this new addition is following the reason of the other, but it is not.

I recommend putting all ERRORs on top with no comments, and OFFs one the bottom with comments.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (rickie) [reply pom.xml]: Added a commit for these changes!

**Comment by rickie** [comment]
Rebased and resolved conflict!

**Review by rnveach — APPROVED** [review]

**Pull request merged**